File: Lab2/multMatrices/clasica/codigo_cpp/clasica.cpp
#include <iostream>
#include <vector>
#include <chrono>

void multiplicarMatrices(int N) {
    std::vector<std::vector<double>> A(N, std::vector<double>(N, 1.0));
    std::vector<std::vector<double>> B(N, std::vector<double>(N, 2.0));
    std::vector<std::vector<double>> C(N, std::vector<double>(N, 0.0));

    auto start = std::chrono::high_resolution_clock::now();

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            for (int k = 0; k < N; k++) {
                C[i][j] += A[i][k] * B[k][j];
            }
        }
    }

    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> duration = end - start;
    
    std::cout << "matriz " << N << "x" << N << " - tiempo: " << duration.count() << " segundos\n";
}

int main() {
    std::cout << "mult clasica en c++\n";

    std::vector<int> tams = {1000};
    
    for(int N : tams) {
        multiplicarMatrices(N);
    }

    return 0;
}

File: Lab2/multMatrices/clasica/codigo_go/clasica.go
package main

import (
	"fmt"
	"time"
)

func multiplicarMatrices(N int) {

	A := make([][]float64, N)
	B := make([][]float64, N)
	C := make([][]float64, N)
	for i := 0; i < N; i++ {
		A[i] = make([]float64, N)
		B[i] = make([]float64, N)
		C[i] = make([]float64, N)
		for j := 0; j < N; j++ {
			A[i][j] = 1.0
			B[i][j] = 2.0
			C[i][j] = 0.0
		}
	}

	start := time.Now()

	for i := 0; i < N; i++ {
		for j := 0; j < N; j++ {
			for k := 0; k < N; k++ {
				C[i][j] += A[i][k] * B[k][j]
			}
		}
	}

	elapsed := time.Since(start)
	fmt.Printf("matriz %dx%d - tiempo: %f segundos\n", N, N, elapsed.Seconds())
}

func main() {
	fmt.Println("mult clasica en Go")

	tams := []int{1000}

	for _, N := range tams {
		multiplicarMatrices(N)
	}
}
